### ISSUE TITLE
Handle floating point imprecision

### ISSUE THREAD

**Issue by jonashaag** [c3js/c3#797]
C3 should take care of floating point rounding imprecision, i.e. values like "0.30000000000000004" (= 0.1 + 0.2), in X and Y axis ticks.

We could
- have a fixed number of decimal places by default
- try to recognize imprecision patterns, like 10 zeros in a row

Someone already reported this in #506, but no action was taken there to resolve the problem.

I think that this should be taken care of in c3, because it's a common problem (anyone dealing with floating point values is exposed to it).

**Comment by dazhbog** [comment]
I had the same issue but I fixed my data, I don't think the graphing library should take care of these things. It should just plot whatever you give it. 

Unless you have a scenario that I have missed.

**Comment by jcollum** [comment]
I disagree. I'm getting some labels that are correct and some that are showing floating point problems:  

![image](https://cloud.githubusercontent.com/assets/704514/15305177/5d094864-1b76-11e6-8573-3a065714c6f7.png)

It's not the data's fault, the data looks like: 

```
{
  _id: "573a3743f6e91aa6b86a881d",
  status: "g",
  dt: 1463433027504,
  time: 2.8683171272278
},
{
_id: "573a374db23965e6eb42ca00",
status: "g",
dt: 1463433037064,
time: 2.8683171272278
},
{
_id: "573a375cb23965e6eb42ca14",
status: "g",
dt: 1463433052072,
time: 2.8683171272278
},
{
_id: "573a376bb23965e6eb42ca20",
status: "g",
dt: 1463433067104,
time: 2.4415690898895
},
{
_id: "573a377ab23965e6eb42ca36",
status: "g",
dt: 1463433082114,
time: 2.4415690898895
},
{
_id: "573a377ff6e91aa6b86a882f",
status: "g",
dt: 1463433087525,
time: 2.4415690898895
},
{
_id: "573a3789b23965e6eb42ca47",
status: "g",
dt: 1463433097093,
time: 2.4415690898895
},
```

This looks like a bug to me, not an enhancement.

**Comment by jcollum** [comment]
@masayuki0812  This looks like a bug to me, not an enhancement.

**Comment by michaelrodov** [comment]
hiw about adding default d3.format?

**Comment by jcollum** [comment]
I don't know if that will work, long-term, since the data coming in can vary a lot (sometimes it's decimal, sometimes it'll be integers).

**Comment by jcollum** [comment]
I tried it: 

![configuration for chart](https://cloud.githubusercontent.com/assets/704514/15327933/0683bd48-1c08-11e6-9ecf-0d5b2f2695db.png)

Still got `0.60000000000001` for my first tick.

**Comment by jcollum** [comment]
I set the y axis format to `d3.format(".3n")` and it seems to be helping. Hard to say though since the error doesn't show up very often.
